OSDOCS-5458: Networking doc improvements for MicroShift (#57678)
openshift/openshift-docs: modules/microshift-blocking-nodeport-access.adoc

// Module included in the following assemblies:
//
// * microshift_networking/microshift-networking.adoc

:_content-type: PROCEDURE
[id="microshift-blocking-nodeport-access_{context}"]

> = Blocking external access to NodePort service on a specific host interface

Reviewer: ```suggestion
= Blocking external access to the NodePort service on a specific host interface
```

> OVN-Kubernetes does not restrict the host interface, therefore NodePort services can be accessed from outside a MicroShift node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer: ```suggestion
OVN-Kubernetes does not restrict the host interface, therefore NodePort services can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.
```

Reviewer: Is it actually NodePort services, or just node ports?

Author: The doc in the MicroShift repo said services, lemme confirm as well

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Author: Right, I was thinking that as well, lemme check with this bc I'm seeing it being used differently around the docs when talking about just NodePort vs the NodePort Service

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer 2: Got it. If this is an exception, maybe we can get some clarity, bring it up at the team meeting, and add it to the glossary.

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer: @kelbrown20 and @michaelryanpeter I just had this issue in a previous PR and my SME (@zshi-redhat) corrected "node ports" to NodePort service, so that's what we have been doing in the Networking portions of the MicroShift docs. https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/networking/microshift-applying-networking-settings

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer 2: That makes sense. According to the repo rules, if we are talking about the specific object, it should be `NodePort` service. But, if this is important to engineering and how customers expect to see it, it should be added to the repo guidelines glossary. Hopefully, that can reduce churn :)

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer 3: Just for reference, I suggested to use `NodePort` to align with the kubernetes service type defined here: https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types, other service types include ClusterIP, LoadBalancer and ExternalName.

> OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Author: Gotcha! Will update!

> OVN-Kubernetes does not restrict the host interface, therefore NodePort services can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

Reviewer 3: ```suggestion
OVN-Kubernetes does not restrict the host interface where a NodePort service can be accessed from outside a {product-title} node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.
```

Reviewer 3: NodePort service is designed to be accessed from outside the cluster node, the `therefore` in above sentence may not be accurate in my opinion.  For security reason, we add this feature to allow user to restrict which host interface the nodeport service can be accessed. hope it makes sense.

Author: Ohh I gotcha! I was misunderstanding the content, thank you!

.Procedure

> . Change the NODEPORT variable to the host port number assigned to your kubernetes NodePort service 

Reviewer 2: See the [repo guidelines](https://redhat-documentation.github.io/supplementary-style-guide/#commands-with-root-privileges) on "run/enter the following command," and please update throughout.

+
[source,terminal]
----
$ export NODEPORT=30700
----

> . Change the INTERFACE_IP value to the the IP address from the host interface that you want to block. 

Reviewer 3: ```suggestion
. Change the INTERFACE_IP value to the IP address from the host interface that you want to block. 
```

+
[source,terminal]
----

> $ export INTERFACE_IP=192.168.150.33

Author: I think they wanted to show the exact IP becuase its referenced later in the doc. And I think the sentence shows that it can be changed. Maybe I can add a `For example` at the end of the sentence to show that its a sample and that the user is meant to change themselves.

> $ export INTERFACE_IP=192.168.150.33

Reviewer 2: Makes sense. Thank you for clarifying.

----

> . Insert a new rule in the `ip-na` table PREROUTING chain to drop the packet and match the destination port and ip. 

Reviewer 3: ```suggestion
. Insert a new rule in the `nat` table PREROUTING chain to drop all packets that match the destination port and ip. 
```

+
[source,terminal]
----
$ sudo nft -a insert rule ip nat PREROUTING tcp dport $NODEPORT ip daddr $INTERFACE_IP drop
----

. List the new rule with the following command:
+
[source,terminal]
----
$ sudo nft -a list chain ip nat PREROUTING
table ip nat {
	chain PREROUTING { # handle 1
		type nat hook prerouting priority dstnat; policy accept;
		tcp dport 30700 ip daddr 192.168.150.33 drop # handle 134
		counter packets 108 bytes 18074 jump OVN-KUBE-ETP # handle 116
		counter packets 108 bytes 18074 jump OVN-KUBE-EXTERNALIP # handle 114
		counter packets 108 bytes 18074 jump OVN-KUBE-NODEPORT # handle 112
	}

> }

Reviewer 2: Please double check your IDE's whitespace handling. This should be fixed throughout. I just noticed the extra line when I made a comment on line 49.

![image](https://user-images.githubusercontent.com/82119979/232599357-f8aab06f-bb1d-4095-ad27-5f83973d11de.png)

Author: Ah, I see it now! Fixing!

----
+
[NOTE]
====

> Note your `handle` number of the newly added rule. You will need the number for removal in the following step. 

Reviewer 2: ```suggestion
Note your `handle` number of the newly added rule. You need to remove `handle` number in the following step.
```
You could also pull this up into the step above. "List the new rule by running the following command. Note your `handle` number ..."

// Module included in the following assemblies:
//
// * microshift_networking/microshift-networking.adoc

:_content-type: PROCEDURE
[id="microshift-blocking-nodeport-access_{context}"]
= Blocking external access to NodePort service on a specific host interface

OVN-Kubernetes does not restrict the host interface, therefore NodePort services can be accessed from outside a MicroShift node. The following procedure explains how to block the NodePort service on a specific host interface and restrict external access.

.Prerequisites 

> //Not sure what prereqs are needed

Author: I couldn't tell if there were prerequisites needed for this procedure. Or rather I couldn't find any

> .Procedure

Reviewer 3: The formating looks good to me, I suggested a few change on the commands.
